hazelcast-client-new/src/main/java/com/hazelcast/client/impl/HazelcastClientInstanceImpl.java

```diff
@@ -135,7 +135,13 @@
     public HazelcastClientInstanceImpl(ClientConfig config) {
         this.config = config;
         final GroupConfig groupConfig = config.getGroupConfig();
-        instanceName = "hz.client_" + id + (groupConfig != null ? "_" + groupConfig.getName() : "");
+        
+        if(config.getInstanceName() != null) {
```

Comment:
Please check all your `if(` changes, the space character is missing everywhere :)